eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py

```diff
@@ -70,218 +69,213 @@ def create_table(private_config_xml_file):
-
-
-def get_assembly_list_for_taxonomy(private_config_xml_file, taxonomy, release_version):
-    assembly_list = set()
+        for tax, asm_acc in get_all_results_for_query(pg_conn, query):
+            insert_entry_for_taxonomy_and_assembly(private_config_xml_file, profile, tax, asm_acc, release_version,
+                                                   ref_dir)
+
+
+def fill_in_from_supported_assembly_tracker(private_config_xml_file, profile, release_version, ref_dir):
+    query = f"""select distinct taxonomy_id as taxonomy, assembly_id as assembly_accession
+                from evapro.supported_assembly_tracker sat"""
+
     with get_metadata_connection_handle("production_processing", private_config_xml_file) as pg_conn:
-        query = f'SELECT assembly_accession from evapro.assembly where taxonomy_id = {taxonomy}'
-        for assembly in get_all_results_for_query(pg_conn, query):
-            assembly_list.add(assembly[0])
-        query = (f"SELECT distinct assembly_accession "
-                 f"from eva_progress_tracker.clustering_release_tracker "
-                 f"where taxonomy={taxonomy} and release_version={release_version} and assembly_accession!='Unmapped'")
-        for assembly in get_all_results_for_query(pg_conn, query):
-            assembly_list.add(assembly[0])
-    return list(assembly_list)
-
-
-def get_filter_group_count_pipeline(taxonomy, assembly_list):
-    filter_by_taxonomy_and_assembly = {
-        "$match":
-            {
-                "tax": {
-                    "$eq": taxonomy,
-                },
-                "asm": {
-                    "$in": assembly_list,
-                }
-            }
-    }
-    group_by_assembly = {
-        "$group":
-            {
-                "_id": {"assembly": "$asm"},
-                "rs_count": {"$sum": 1},
-            }
-    }
-    return [filter_by_taxonomy_and_assembly, group_by_assembly]
-
-
-def get_rs_count_per_assembly_in_collection(mongo_source, coll_name, pipeline):
-    coll = mongo_source.mongo_handle[mongo_source.db_name][coll_name]
-    agg_result = coll.aggregate(pipeline, batchSize=0)
-    agg_res_dict = {}
-    for entry in agg_result:
-        agg_res_dict[entry['_id']['assembly']] = entry['rs_count']
-    return agg_res_dict
-
-
-def get_sources_and_rs_count(cve_res, dbsnp_res, assembly):
-    if assembly in cve_res and assembly in dbsnp_res:
-        return ['DBSNP, EVA', cve_res[assembly] + dbsnp_res[assembly]]
-    elif assembly in cve_res and assembly not in dbsnp_res:
-        return ['EVA', cve_res[assembly]]
-    elif assembly not in cve_res and assembly in dbsnp_res:
-        return ['DBSNP', dbsnp_res[assembly]]
+        for tax, asm_acc in get_all_results_for_query(pg_conn, query):
+            insert_entry_for_taxonomy_and_assembly(private_config_xml_file, profile, tax, asm_acc, release_version,
+                                                   ref_dir)
```

Comment:
This currently will only support `source='EVA'`. I think when we remap variants that are in DBSNP then these variant will go in the DBSNP collections
You should retrieve the source based on source of the other assembly of that species.

Reply:
updated